processor/tasks.go

```diff
@@ -137,14 +85,50 @@ func (task *TitleTask) Execute(w *worker.Worker) error {
 	return task.executeTask(w, task.proc, task.processTitle)
 }
 
+// processTroubleState will check if the queue item is troubled, and if so, will
+// query the trouble for information about how the user wishes it to be resolved.
+// This method will return an error if the processing fails. The second return (bool)
+// indicates whether or not the item has been fully processed. Item trouble is cleared
+// when this method returns
+func (task *TitleTask) processTroubleState(queueItem *QueueItem) (error, bool) {
+	defer queueItem.ClearTrouble()
+
+	if queueItem.Trouble != nil {
+		if trblCtx := queueItem.Trouble.ResolutionContext(); trblCtx != nil {
+			// Check for the mandatory 'info' key.
+			info, ok := trblCtx["info"]
+			if !ok {
+				return errors.New("resolution context is missing 'info' key as is therefore invalid - ignoring context!"), false
+			}
+
+			// Assign the 'info' provided as the items TitleInfo and move on.
+			titleInfo, ok := info.(TitleInfo)
+			if ok {
+				queueItem.TitleInfo = &titleInfo
+				task.advance(queueItem)
+
+				return nil, true
+			}
+
+			return errors.New("resolution contexts 'info' key contains an invalid value! Failed to cast to 'TitleInfo'"), false
+		}
+	}
+
+	return nil, false
```

Comment:
This needs changing - if we get here it means something went wrong logically. Only line 110 should be a non-error return, any other code flow indicates an issue.